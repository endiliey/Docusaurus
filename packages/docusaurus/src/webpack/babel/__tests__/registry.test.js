/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
const path = require('path');
const {transformFileSync} = require('@babel/core');

const plugin = require.resolve('../registry');

test('transform registry', () => {
  const {code} = transformFileSync(
    path.join(__dirname, '__fixtures__', 'registry', 'simple.js'),
    {
      plugins: ['@babel/syntax-dynamic-import', plugin],
      babelrc: false,
      configFile: false,
    },
  );
  expect(code).toMatchSnapshot();
});

test('only add webpackChunkName comment', () => {
  const {code} = transformFileSync(
    path.join(__dirname, '__fixtures__', 'registry', 'simple.js'),
    {
      plugins: [
        '@babel/syntax-dynamic-import',
        [plugin, {onlyChunkName: true}],
      ],
      babelrc: false,
      configFile: false,
    },
  );
  expect(code).toMatchSnapshot();
});

test('does not transform file without registry pragma', () => {
  const {code} = transformFileSync(
    path.join(__dirname, '__fixtures__', 'registry', 'no-pragma.js'),
    {
      plugins: ['@babel/syntax-dynamic-import', plugin],
      babelrc: false,
      configFile: false,
    },
  );
  expect(code).toMatchSnapshot();
});
