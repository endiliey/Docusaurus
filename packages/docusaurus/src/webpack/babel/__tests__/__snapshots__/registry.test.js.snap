// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not transform file without registry pragma 1`] = `
"export default {
  'component---site-src-pages-help-js-7-e-3-222': {
    loader: () => import('@site/src/pages/help.js')
  }
};"
`;

exports[`only add webpackChunkName comment 1`] = `
"/*  */
export default {
  'component---site-src-pages-help-js-7-e-3-222': {
    loader: () => import('@site/src/pages/help.js')
  },
  'component---site-src-pages-index-jsc-4-f-f99': {
    loader: () => import('@site/src/pages/index.js')
  },
  'component---site-src-pages-feedback-index-js-81-e-a42': {
    loader: () => import('@site/src/pages/feedback/index.js')
  }
};"
`;

exports[`transform registry 1`] = `
"/*  */
export default {
  'component---site-src-pages-help-js-7-e-3-222': {
    loader: () => import('@site/src/pages/help.js')
  },
  'component---site-src-pages-index-jsc-4-f-f99': {
    loader: () => import('@site/src/pages/index.js')
  },
  'component---site-src-pages-feedback-index-js-81-e-a42': {
    loader: () => import('@site/src/pages/feedback/index.js')
  }
};"
`;
